test/types/type-provider.test-d.ts

```diff
@@ -0,0 +1,129 @@
+import fastify, { FastifyTypeProvider } from '../../fastify'
```

Comment:
Can you add some assertions to other handlers? I mean:

```ts
preHandler: (req, res) => {
  expectType<X>(req.body)
}
```
```

Reply:
Have added an additional test for each request hook.